cmd/incus/snapshot.go

```diff
@@ -157,6 +175,13 @@ func (c *cmdSnapshotCreate) Run(cmd *cobra.Command, args []string) error {
 
 	if c.flagNoExpiry {
 		req.ExpiresAt = &time.Time{}
+	} else if !stdinData.ExpiresAt.IsZero() {
+		if stdinData.ExpiresAt.Before(time.Now()) {
+			return fmt.Errorf(i18n.G("Expiry date cannot be in the past"))
+		} else {
+			req.ExpiresAt = &stdinData.ExpiresAt
+			fmt.Printf(req.ExpiresAt.String())
+		}
```

Comment:
A bit confused about why we have that stuff here, we shouldn't perform input validation on the client side, it's really for the server to handle instead.

If the user provides us a YAML input with a data that's past expiry, it will cause the server to expire the snapshot on the next expiry run, but that's fine, that may actually be useful for testing.

Reply:
Gotcha, I'll make a fixup!